eos_api.hpp

```diff
@@ -1341,377 +1332,343 @@ uint32_t oraclize_getSenderNonce(account_name sender)
     return nonce;
 }
 
-checksum256 oraclize_getCommitment(uint64_t shortqueryid)
-{
-    ds_scommitment last_commitments(N(_self), N(_self));
-    auto itr = last_commitments.find(shortqueryid);
-    checksum256 commitment;
-    if (itr != last_commitments.end())
-    {
-        commitment = itr->commitment;
-    }
-    return commitment;
-}
-
-checksum256 oraclize_getQueryId(uint64_t shortqueryid)
-{
-    ds_scommitment last_commitments(N(_self), N(_self));
-    auto itr = last_commitments.find(shortqueryid);
-    checksum256 queryid;
-    if (itr != last_commitments.end())
-    {
-        queryid = itr->queryid;
-    }
-    return queryid;
-}
-
-checksum256 oraclize_getNextQueryId(account_name sender)
+capi_checksum256 __oraclize_getNextQueryId(const name sender)
 {
     // Get values to generate the queryId
-    uint32_t nonce = oraclize_getSenderNonce(sender);
-    size_t tx_size = transaction_size();
+    const uint32_t nonce = __oraclize_getSenderNonce(sender);
+    const size_t tx_size = transaction_size();
     // calculate the hash of the previous values
-    uint8_t tbh[sizeof(sender) + sizeof(nonce) + sizeof(tx_size)]; //account_name[uint64_t] + nonce[uint32_t]
+    uint8_t tbh[sizeof(sender) + sizeof(nonce) + sizeof(tx_size)];
     std::memcpy(tbh, &sender, sizeof(sender));
     std::memcpy(tbh + sizeof(sender), &nonce, sizeof(nonce));
     std::memcpy(tbh + sizeof(sender) + sizeof(nonce), &tx_size, sizeof(tx_size));
-    checksum256 calc_hash;
+    capi_checksum256 calc_hash;
     sha256((char *)tbh, sizeof(tbh), &calc_hash);
     return calc_hash;
 }
 
-checksum256 oraclize_getQueryId_local__(checksum256 queryId, account_name sender)
+capi_checksum256 __oraclize_getQueryId_local(const capi_checksum256 queryId, const name sender)
```

Comment:
make this function more comprehensive: it should check that the queryId being passed matches with the one in the local table, return true/false accordingly - new function name could be "__oraclize_queryId_match"